src/windows/mod.rs

```diff
@@ -81,19 +84,27 @@ impl SystemReaderInterface for SystemReader {
     }
 
     fn get_hosts_file(&self) -> Option<String> {
-        let mut s = String::new();
-        let mut t = env::var("SYSTEMROOT").unwrap_or("".to_string()).to_string();
-        t.push_str(&"\\system32\\drivers\\etc\\hosts".to_string());
-        File::open(t).ok()?.read_to_string(&mut s).ok()?;
-        Some(s)
+        let mut string = String::new();
+        let mut file_location = env::var("SYSTEMROOT").unwrap_or("".to_string()).to_string();
+        file_location.push_str(&"\\system32\\drivers\\etc\\hosts".to_string());
```

Comment:
replace with `file_location.push_str("\\system32\\drivers\\etc\\hosts");` to avoid a string allocation.